### ISSUE TITLE
Support colorized tree output for Automation API operations

### ISSUE THREAD

**Issue by bison** [pulumi/pulumi#11681]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

I make heavy use of the Pulumi Automation API to the point that it basically drives all of our Pulumi operations, including as part of a custom CLI tool. Unfortunately, with my custom CLI tool and the Automation API, I don't have access to the nice colorized tree output that the `pulumi` CLI provides for stack operations. I would love to be able to optionally get that nice output when using the Automation API. 

### Affected area/feature

- Automation API

**Comment by RobbieMcKinstry** [comment]
Hi @bison very cool that you're using Pulumi so heavily! I love to hear stories like your's.

I took a brief look through the internals and the API. I think I found something that might suit your needs. Can you confirm whether this will work for you?

I see on `preview`, there's a `previewOption` to force colored output ([link](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/automation/#PreviewOptions-color)). If set, then the `previewResult` should colorize stdout, which is returned from preview ([link](https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/automation/#PreviewResult-stdout)). The same is true for `up`.

I just checked the NodeJS API, but I know there's some inconsistency between our language support for Automation API. Please let me know if this works for you, or doesn't work. Happy holidays!

**Comment by bison** [comment]
Hi @RobbieMcKinstry, thanks for the response. I'm using Go, which does not have a an option function to set the `color` field. I can make one myself, and set `color: "always"`. Then I *do* get color output, which is nice, but I still don't get the nice tree view that I would with the Pulumi CLI.

Ultimately what I want is the ability to stream events from the Automation API to the same code the Pulumi CLI uses: [pkg/backend/display](https://github.com/pulumi/pulumi/tree/master/pkg/backend/display) -- which isn't currently possible. I'd also be happy to help implement this if it's something that would be accepted.

**Comment by bison** [comment]
Yep, that works to get colorized output, but you still can't get the nice animated tree style output that you get from the CLI.
